File: code-examples/java-rs/connection/stable-api-connect.java
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.ConnectionString;
import com.mongodb.MongoClientSettings;
import com.mongodb.ServerApiVersion;
import com.mongodb.ServerApi;
import com.mongodb.reactivestreams.client.MongoDatabase;
import org.bson.Document;
import com.mongodb.client.result.UpdateResult;
import reactor.core.publisher.Mono;

public class Main {
    public static void main(String[] args) {
        // Replace the placeholder with your Atlas connection string
        ConnectionString connectionString = new ConnectionString("<connection string>");

        MongoClientSettings settings = MongoClientSettings.builder()
                .applyConnectionString(connectionString)
                .serverApi(new ServerApi.Builder()
                        .version(ServerApiVersion.V1)
                        .strict(true)
                        .deprecationErrors(true)
                        .build())
                .build();
        // Create a MongoClient with a MongoClientMetrics object to set the Stable API version
        MongoClient mongoClient = MongoClients.create(settings);

        try {
            MongoDatabase database = mongoClient.getDatabase("admin");

            Mono.from(database.runCommand(new Document("ping", 1)))
                    .doOnSuccess(doc -> System.out.println("Pinged your deployment. You successfully connected to MongoDB!"))
                    .doOnError(error -> System.out.println("An error occurred: " + error))
                    .block();
        } finally {
            mongoClient.close();
        }
    }
}

File: code-examples/scala/connection/stable-api-connect.scala
import org.mongodb.scala._

import scala.concurrent.ExecutionContext.Implicits.global
import scala.util.{Failure, Success}

object MongoDBScalaClient { 
  def main(args: Array[String]): Unit = {
    
    val settings: MongoClientSettings = MongoClientSettings.builder()
      .applyConnectionString(new ConnectionString("<connection string>"))
      .serverApi(new ServerApi.Builder().version(ServerApiVersion.V1).strict(true).deprecationErrors(true).build())
      .build()

    val mongoClient: MongoClient = MongoClient(settings)
     
    val db = mongoClient.getDatabase("admin")
    
    val command = new org.bson.Document("ping", 1)
    db.runCommand(command).toFuture().onComplete {
      case Success(response) => println("Pinged your deployment. You successfully connected to MongoDB!")
      case Failure(e) => println(s"An error has occured: $e")
    }
    
    sys.addShutdownHook {
      mongoClient.close()
    }
  }
}
